### PR TITLE
Add MoGe-3 monocular geometry estimation model

### PR DESCRIPTION
**Pull request by eklipse2k8**
Port MoGe-3 (moge.model.v3) to MLX: DINOv2 encoder with output projections, convolutional neck + points/mask/normal/scale heads, and the sparse 3D UNet refiner.

The refiner normally depends on FlexGEMM (Triton, CUDA-only); here the submanifold convolution, mean pooling, and nearest upsample are re-implemented in pure MLX with sort + binary-search + gather, so the model runs on Apple silicon. Transposed convolutions are computed as exact per-pixel GEMMs followed by a 2x2 pixel shuffle, and all resizes match torch F.interpolate (including antialiased bilinear and the scale-factor bicubic position-embedding path).

Adds moge3 model package (config, vision, conv_stack, sparse, refiner, geometry, generate, processor, checkpoint converter, README) and a TestMoge3 test class.

Converted checkpoints uploaded to mlx-community/moge-3-vitl-mlx-fp32 and mlx-community/moge-3-vitg-mlx-fp32. Validated against the torch reference (CPU, fp32, pure-torch FlexGEMM shim): identical valid-pixel masks, <0.1% median relative depth error at default settings, ~2.1s per 800x547 image on GPU.

### CODE REVIEW

**Comment by Lazarus-931** [comment]
hi @eklipse2k8, thanks for the pr!

lgtm!

**Review by Lazarus-931 — APPROVED** [review]

File: mlx_vlm/models/dinov2_backbone.py [hunk 1]
```diff
@@ -1,49 +1,33 @@
-"""DINOv2 vision backbone for Video Depth Anything (channel-last MLX port)."""
+"""Shared DINOv2 vision transformer (channel-last MLX port).
```
Comment (lucasnewman) [inline mlx_vlm/models/dinov2_backbone.py]: I think this should stay in the original location or be moved to a `dinov2` directory -- we typically don't have free-floating model classes in the `models` root.

**Review by lucasnewman — CHANGES_REQUESTED** [review]
@eklipse2k8 This looks really cool! Take a look at my comment about moving the backbone into its own directory -- if we resolve that we can merge.

I think it might be worth considering how we could make a unified interface for depth prediction models if you're planning to continue adding models in this family. Not required for this PR but just something to think about.

**Comment by eklipse2k8** [comment]
Okay, I started a port of dinov2 and moved over the beginning bits. It has a huge family of vision models based on it, so I think it's worth the separation. I looked at the official repo for it, and it's large enough that it could use its own full port separate from this.

**Review by lucasnewman — APPROVED** [review]
Thanks!
